### ISSUE TITLE
Field incorrectly marked as unused

### ISSUE THREAD

**Issue by sunwanxin213** [microsoft/TypeScript#49363]
Issue Type: <b>Feature Request</b>

``` javascript
function fn(cb) {
    return cb;
}

class Test {
    #abc = 123;

    #bcd = 456;

    check() {
        console.log(this.#abc);

        Test.prototype.check = fn(function () {
            console.log(this.#bcd);
        });

        return Test.prototype.check.call(this);
    }
}
```

#abc is ok
#bcd is wrong

VS Code version: Code 1.67.2 (c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5, 2022-05-17T18:15:52.058Z)
OS version: Windows_NT x64 10.0.22000
Restricted Mode: No

![QQ截图20220603062919](https://user-images.githubusercontent.com/3002658/171748839-9f475e85-9a0b-491b-b0c6-780dd3bb0737.png)

and can run

![QQ截图20220603062933](https://user-images.githubusercontent.com/3002658/171748867-20068b4c-a7e9-4fe2-b30c-e51ddaf4868a.png)

**Comment by mjbvz** [comment]
This is caused because we can't infer what `this` should be inside the function because the code is too dynamic

Likely not worth trying to fix

**Comment by mjbvz** [comment]
Try setting `editor.showUnused: false`

**Comment by sunwanxin213** [comment]
thank you

**Comment by sunwanxin213** [comment]
Provide an idea, if the current file can match the whole word, it has been used or can this situation be alleviated? After all, it's a private domain.
If your team thinks the issue can be ignored, you can close the issue.
Finally, thanks for your reply.

**Comment by RyanCavanaugh** [comment]
TS can't see the correct `this` type because of the call to `fn` "in the way". You can annotate with a JS Doc to re-fix it:
```ts
function fn(cb) {
    return cb;
}

class Test {
    #abc = 123;

    #bcd = 456;

    check() {
        console.log(this.#abc);

        Test.prototype.check = fn(
        /**
         * @this Test
         */
        function () {
            console.log(this.#bcd);
        });

        return Test.prototype.check.call(this);
    }
}
```
